nucleus/cluster/admin/src/main/java/fish/payara/admin/cluster/AddInstanceToDeploymentGroupCommand.java

```diff
@@ -119,42 +115,54 @@ public void execute(AdminCommandContext context) {
             return;
         }
 
-        // OK set up the reference
-        try {
-            ConfigSupport.apply((DeploymentGroup dg1) -> {
-                DGServerRef ref = dg1.createChild(DGServerRef.class);
-                ref.setRef(instanceName);
-                dg1.getDGServerRef().add(ref);
-                return ref;
-            }, dg);
-
-        } catch (TransactionFailure e) {
-            report.setMessage("Failed to add instance to deployment group");
-            report.setActionExitCode(ActionReport.ExitCode.FAILURE);
-            report.setFailureCause(e);
-        }
-        
-        // now run the command to add application ref to the instance
-        for (ApplicationRef applicationRef : dg.getApplicationRef()) {
-            CommandInvocation inv = commandRunner.getCommandInvocation("create-application-ref", report, context.getSubject());
-            ParameterMap parameters = new ParameterMap();
-            parameters.add("target", instanceName);
-            parameters.add("name", applicationRef.getRef());
-            parameters.add("virtualservers", applicationRef.getVirtualServers());
-            parameters.add("enabled", applicationRef.getEnabled());
-            parameters.add("lbenabled", applicationRef.getLbEnabled());
-            inv.parameters(parameters).execute();
-        }
-        
-        // for all resource refs add resource ref to instance
-        for (ResourceRef resourceRef : dg.getResourceRef()) {
-            CommandInvocation inv = commandRunner.getCommandInvocation("create-resource-ref", report, context.getSubject());
-            ParameterMap parameters = new ParameterMap();
-            parameters.add("target", instanceName);
-            parameters.add("reference_name", resourceRef.getRef());
-            parameters.add("enabled", resourceRef.getEnabled());
-            inv.parameters(parameters).execute();
-            
+        final List<String> instances = new ArrayList<String>(Arrays.asList(instanceName.split(",")));
```

Comment:
Does the list need to be final?

Also, the second generic parameter isn't needed. This could probably be something like:

    List<String> instances = new ArrayList<>(asList(instanceName.split(",")));

But then again, the new ArrayList is likely not needed either, so may be:

    List<String> instances = asList(instanceName.split(","));